include/usb/usb_ch9.h

```diff
@@ -0,0 +1,246 @@
+/*
+ * Copyright (c) 2020 PHYTEC Messtechnik GmbH
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief USB Chapter 9 structures and definitions
+ *
+ * This file contains the USB Chapter 9 structures definitions.
+ */
+
+#include <version.h>
+#include <sys/util.h>
+
+#ifndef ZEPHYR_INCLUDE_USB_CH9_H_
+#define ZEPHYR_INCLUDE_USB_CH9_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+struct usb_req_type_field {
+#if __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__
```

Comment:
This is not portable, some discussion at https://stackoverflow.com/questions/1490092/c-c-force-bit-field-order-and-alignment
I propose a getter/setter macro/func for this instead.

Reply:
We use this in the BLE Controller as well, and we discussed whether to use the same approach here or not. We are still to find a single compiler that doesn't do the right thing with bitfields and `__BYTE_ORDER__` macros, so I really am not sure this is a bad idea.

Reply:
Probably this works most of the time, but it is still not portable. Just wondering if MISRA-C has some rule against these.